flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGateTest.java

```diff
@@ -542,6 +543,29 @@ public void testUpdateUnknownInputChannel() throws Exception {
 		}
 	}
 
+	@Test
+	public void checkInputGateRemoveInNetworkEnvironment() throws IOException {
+		NetworkEnvironment network = createNetworkEnvironment();
+		InputChannelDeploymentDescriptor[] channelDescs = new InputChannelDeploymentDescriptor[]{
+			// Local
+			new InputChannelDeploymentDescriptor(
+				new ResultPartitionID(),
+				ResultPartitionLocation.createLocal())};
+		IntermediateDataSetID id = new IntermediateDataSetID();
+		InputGateDeploymentDescriptor igdd = new InputGateDeploymentDescriptor(id,
+			ResultPartitionType.PIPELINED, 0, channelDescs);
+		SingleInputGate[] inputGates = network.createInputGates("",
+			new NoOpTaskActions(), Collections.singletonList(igdd),
+			new UnregisteredMetricsGroup(), new UnregisteredMetricsGroup(), new UnregisteredMetricsGroup(),
+			new SimpleCounter());
+		assertEquals(1, inputGates.length);
+		assertEquals(1, network.getInputGatesById().size());
+		assertTrue(network.getInputGatesById().containsKey(id));
+		inputGates[0].close();
+		assertEquals(0, network.getInputGatesById().size());
+		assertFalse(network.getInputGatesById().containsKey(id));
```

Comment:
The two last assertions seem to be redundant.

Reply:
I will rewrite a bit the test to make it more generic